### PR TITLE
Functional test python  aasemble django build test

### PR DESCRIPTION
**Pull request by anand0707**
This test verifies whether a build is started after a "Package Source" detail is submitted from UI.
It fills git_url, branch and series and checks whether build has started on UI after pressing "build" button,
#### 

This pull request also includes a fix found from above test.
Issue:
Split method does not work for binary strings in Python 3.4. The strings starting with prefix b' are considered binary strings on these strings "split" method does not work and gives the error.
"Type str doesn't support the buffer API"

I am facing this issue on Poll method of Model.py
### 

Fix:
def Poll(self):
......
stdout = run_cmd(cmd)
stdout = stdout.decode() # This is fix for python 3.4. This line has no effect for python 2.7
.......

### CODE REVIEW

**Comment by sorenisanerd** [comment]
I'm a bit confused about this test.

On the surface, it looks like an end-to-end test of a particular workflow, but you're replacing critical elements in the middle: For a start, you're not using Celery's periodic tasks and you're calling poll_one in a different way than it happens in production.

You might as well skip the celery part entirely.

Just register the package source and call poll_one directly (no .apply_async(), no .delay(), nothing like that) and verify that the build shows in the web ui. You would also avoid the race condition: What if Celery didn't get a chance to run the poll_one by the time when you fetch the build list ui?

**Comment by anand0707** [comment]
Made some changes as per comments 
1. Now calling Poll_one without any special call.
2. Removed CELERY_EAGER_PROPAGATES_EXCEPTIONS and BROKER_BACKEND settings over_ride

Now just one setting is being overridden i.e CELERY_ALWAYS_EAGER  ( without it , it is not able to see the build)

**Comment by sorenisanerd** [comment]
I don't see how Celery is involved. Other tests call these methods just fine. Please don't give up. Keep digging, figure out why this is needed.

File: aasemble/django/apps/buildsvc/tests_functional.py [hunk 1]
```diff
@@ -74,6 +76,42 @@ def test_source_package(self):
...
+        self.selenium.get(self.live_server_url)
+        self.selenium.add_cookie(session_cookie)
+        # test whether sources page opens after user logs in
+        self.selenium.get('%s%s' % (self.live_server_url, '/buildsvc/sources/'))
+        self.selenium.set_window_size(1024, 768)
+        self.sources_button.click()
+        git_url = "https://github.com/aaSemble/python-aasemble.django.git"
+        self.create_new_package_source(git_url=git_url, branch='master', series='brandon/aasemble')
+        from .models import PackageSource
+        PackageSourceList = PackageSource.objects.all()
```
Comment (sorenisanerd) [inline aasemble/django/apps/buildsvc/tests_functional.py]: `PackageSource.objects.filter(git_url=git_url)` does what you want here.

**Comment by anand0707** [comment]
Okay..We are calling tasks ( like poll_one, build ) with .delay to achieve asynchronous call, that is where Celery comes to into the scene. Celery takes the responsibility of scheduling these tasks with their asyncResult objects.

So in other words, If we are calling tasks as task.delay either we need Celery worker in background or we can choose CELERY_ALWAYS_EAGER = True  .it will override whole scheduling and we will be able make delay and direct calling same,
For example: 
task(args)
and task.delay(args) with be the same.

Some Reference from web:  Eager scheduling section of https://www.caktusgroup.com/blog/2014/06/23/scheduling-tasks-celery/

**Comment by sorenisanerd** [comment]
Good work figuring out the Celery stuff. Thanks!

**Pull request merged**